src/newgrf_townname.cpp

```diff
@@ -21,69 +21,45 @@
 
 #include "safeguards.h"
 
-static GRFTownName *_grf_townnames = nullptr;
+static std::vector<GRFTownName> _grf_townnames;
 static std::vector<StringID> _grf_townname_names;
```

Comment:
how much effort to get rid of this global? seems like it's entirely contained within `_grf_townnames`

Reply:
_grf_townnames is per GRF, but _grf_townname_names isn't, so it would need more restructuring.